packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -364,6 +364,19 @@ class OAuthClient extends CozyStackClient {
     return new AccessToken(result)
   }
 
+  /**
+   * Fetches a new session code. Only usable by the Flagship application
+   *
+   * @throws {NotRegisteredException} When the client isn't certified to be the Flagship application
+   * @returns {Promise} A promise that resolves with a new session_code
+   */
+  async fetchSessionCode() {
+     return this.fetchJSON(
```

Comment:
not sure you have a direct access to fetchJSON.

Instead
```
this.stackClient.fetchJSON
```

should work

Reply:
according to the implementation `class OAuthClient extends CozyStackClient `, `fetchJSON` should  be an inherited public method. Other methods in OAuthClient use this.fetchJSON() directly

Reply:
Oh. We're in OAuthClient.js ! It's worse. 

OAuth client is there to handle... OAuthClient "connexion". Here this is not what you're doing. 

You are manipulating the OAuthClient doctype. You should definitively use `OAuthClientsCollection` ;)

Reply:
In fact I don't know exactly... 

You're not manipulating a doctype since you're calling the /auth/ route. And you're just creating a new code, not something we want to store in the redux store... 

OauthClient.js could be the right place, let me think about it

Reply:
Let's keep it there. 

Can you fix the CI please

Reply:
@Crash--  it is done